### DISCUSSION TITLE
Machine-oriented input format?

### DISCUSSION THREAD

**Discussion by akx** [typst/typst#4908] [Ideas]
I'd love to use Typst for laying out machine-generated content, but as it stands, the Typst markup language doesn't seem trivial to emit.

Something like a stream of (ND)JSON could maybe work here, where e.g. (and I haven't thought this through!)

```typst
#set par(justify: true)

= Background
In the case of glaciers, fluid...
```
could become
```json
["#set", "par", {"justify": true}]
["heading", "Background"]
"In the case of glaciers, fluid..."
```

or
```typst
Total displaced soil by glacial flow:

$ 7.32 beta +
  sum_(i=0)^nabla Q_i / 2 $
```
could become
```json
"Total displaced soil by glacial flow:"
["Math", "7.32 beta + sum_(i=0)^nabla Q_i / 2"]
```

Is this a silly thought?

**Comment by laurmaedje** [comment]
What's typically a better idea is to instead emit your logical data as JSON or similar and ingest that in Typst via the data loading functions. Then you can handle the data generation in whatever language while using Typst's programming capabilities for the typesetting part.
